app/src/main/java/com/example/basiccalculator/MainActivity.java

```diff
@@ -72,7 +72,7 @@ else if(buttonText.equals("=")){
             result.setText(data.calculatedResult(finalResult));
         }
         else if(buttonText.equals("C")){
-            if(dataToCalculate.equals("0")){
+            if(dataToCalculate.equals("0") || dataToCalculate.equals("") ){
```

Comment:
1) instead of using buttons, you need to use recycler view or grid view. 
2) Folder structure should be , ui, model, data, 
ui should include all your Fragments and Activity. 
model should have APIs; 
data should have data needed for your app. Right now data needed is symbols and number of each button. 
you should not hard code. Your view should have number part and symbol part. Yon need to call getNumber and get Symbols using live data